python/ray/air/session.py

```diff
@@ -0,0 +1,233 @@
+import abc
+from enum import Enum, auto
+import logging
+from typing import Optional, Dict
+
+from ray.air.checkpoint import Checkpoint
+
+logger = logging.getLogger(__name__)
+
+
+class SessionType(Enum):
+    TRAIN = auto()
+    TUNE = auto()
+
+
+class Session(abc.ABC):
+    """The canonical session interface that both Tune and Train session implements.
+
+    User can interact with this interface to get session information,
+    as well as reporting metrics and saving checkpoint.
+    """
+
+    @property
+    @abc.abstractmethod
+    def type(self) -> SessionType:
+        """The type of the session.
+
+        Returns:
+            The type of the session.
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def report(self, metrics: Dict, checkpoint: Optional[Checkpoint] = None) -> None:
+        """Report metrics and optionally save checkpoint.
+
+        The checkpoint is guaranteed to be saved onto Driver node.
```

Comment:
Whether it's saved on the driver node or not is dependent on what's configured in `CheckpointStrategy` right?

Reply:
In my `CheckpointStrategy` if I disable checkpoints, then I don't want this written to disk on the head node.